src/client/Microsoft.Identity.Client/MsalErrorMessage.cs

```diff
@@ -221,6 +221,7 @@ internal static class MsalErrorMessage
             "ConfidentialClientApplication implementation does not implement IConfidentialClientApplicationExecutor.";
 
         public const string ClientCredentialAuthenticationTypesAreMutuallyExclusive = "ClientSecret, Certificate and ClientAssertion are mutually exclusive properties. Only specify one. See https://aka.ms/msal-net-client-credentials";
+        public const string ClientCredentialAuthenticationTypeMustBeDefined = "One client credential type required either: ClientSecret, Certificate OR ClientAssertion must be defined when creating a Confidential Client. Only specify one. See https://aka.ms/msal-net-client-credentials";
```

Comment:
I think this will read better as "One of the required client credential types, either a ClientSecret, Certificate or ClientAssertion must be defined when creating a Confidential Client. Only specify one. See [https://aka.ms/msal-net-client-credentials](https://aka.ms/msal-net-client-credentials)";

Reply:
> LGTM.
> 
> I've proposed some suggestions using the "suggestions" in PRs mechanism in PRs which @mmacy showed me a couple of days ago (which you can approve directly or not)
> 
> ![image](https://user-images.githubusercontent.com/13203188/80696896-67f68680-8ad8-11ea-9227-93c971cc2c83.png)

I wonder if there is a way to make this work with code flow?